sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DistinctTest.java

```diff
@@ -251,4 +253,64 @@ public void testTriggeredDistinctRepresentativeValues() {
         KV.of(1, "k1"), KV.of(2, "k2"), KV.of(3, "k3"));
     triggeredDistinctRepresentativePipeline.run();
   }
+
+  /**
+   * Regression test: when all values are emitted by a speculative trigger, caused a null KV
+   * when the on-time firing occurred.
+   */
+  @Test
+  @Category({ValidatesRunner.class, UsesTestStream.class})
+  public void testTriggeredDistinctRepresentativeValuesEmpty() {
+    Instant base = new Instant(0);
+    TestStream<KV<Integer, String>> values =
+        TestStream.create(KvCoder.of(VarIntCoder.of(), StringUtf8Coder.of()))
+            .advanceWatermarkTo(base)
+            .addElements(TimestampedValue.of(KV.of(1, "k1"), base))
+            .advanceProcessingTime(Duration.standardMinutes(1))
+            .advanceWatermarkToInfinity();
+
+    PCollection<KV<Integer, String>> distinctValues = triggeredDistinctRepresentativePipeline
+        .apply(values)
+        .apply(Window.<KV<Integer, String>>into(FixedWindows.of(Duration.standardMinutes(1)))
```

Comment:
With Java8 the type can probably be inferred now.

Reply:
Unfortunately not :-/